### DISCUSSION TITLE
Incorrect relative and weighted download/ install statistics with packages associated to dependencies

### DISCUSSION THREAD

**Discussion by ganeshkbhat** [npm/feedback#798] [Registry]
Incorrect relative and weighted download/ install statistics with packages associated to dependencies:

I do not understand the searches metrics so much since I have ot done a deep dive in it. I wish not to comment on that. But here is a question I am raising on the basic download stats:

- I have two packages (both in development stage): [cgijs](https://www.npmjs.com/package/cgijs), [phpcgijs](https://www.npmjs.com/package/phpcgijs).
- The package phpcgjs@1.0.0-beta8 has a dependency cgijs@1.1.0. cgijs@1.1.0 has bee published sometime back. However, phpcgijs@v1.0.0-beta8 has been published a few days (a week) back with a dependency of cgijs. 
- The issue: If I download phpcgijs@v1.0.0-beta8, it has to download cgijs@v1.1.0. Considering last weeks download of 30 for phpcgijs@1.0.0.-beta8, I yet have cgijs@v1.1.0 at 12 downloads a week. (19th to 25th October 2022)
- I am expecting cgijs@v1.1.0 downloads to be at downloads@cgijs@v1.1.0 + downloads@phpcgijs@1.0.0-beta8

This seems to be a gross actual download stats issue. All other stats of downloads and stats based on downloads depends on this calculation. I do not undertand why this was not noticed. I wonder why?

![Web capture_26-10-2022_15300_www npmjs com](https://user-images.githubusercontent.com/3424344/197998416-e20da24b-b4a9-498a-a3a0-c3ff5bac9dc9.jpeg)
![Web capture_26-10-2022_153013_www npmjs com](https://user-images.githubusercontent.com/3424344/197998453-7a8a06ad-cf68-4a08-86f5-076c5c067bb5.jpeg)

![Web capture_26-10-2022_161848_npm-stat com](https://user-images.githubusercontent.com/3424344/198033493-615d4b94-c0a0-4f28-9dd3-0b848e55c18c.jpeg)
![Web capture_26-10-2022_161820_npm-stat com](https://user-images.githubusercontent.com/3424344/198033502-194c28dc-9dc6-402a-8034-7995febec695.jpeg)

**Comment by ganeshkbhat** [comment]
https://github.com/npm/feedback/discussions/66#discussioncomment-3969886

**Comment by ljharb** [comment]
This could easily happen if the transitive dep was cached and the other one wasn’t. It’s not a reasonable expectation that these downloads match up.

**Reply by ganeshkbhat** [reply]
This was not cache issue. cache issues work same and will behave the same with both packages similarly over a period of time. so i believe the point is irrelevant

**Reply by ljharb** [reply]
I mean local cache - someone installing the parent might already have the child via another install.

**Reply by ganeshkbhat** [reply]
> I mean local cache - someone installing the parent might already have the child via another install.

Possible. I doubt local cache will be the case for anything more than 15-20 downloads in 30 downloads of a dependant package. 50% variation due to local cache is a big deal. my point is valid considering anything more than 27 is a valid dataset.
